libraries/Zigbee/examples/Zigbee_Illuminance_Sensor/Zigbee_Illuminance_Sensor.ino

```diff
@@ -0,0 +1,141 @@
+// Copyright 2025 Espressif Systems (Shanghai) PTE LTD
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @brief This example demonstrates Zigbee illuminance sensor.
+ *
+ * The example demonstrates how to use Zigbee library to create a end device illuminance sensor.
+ * The illuminance sensor is a Zigbee end device, which is controlled by a Zigbee coordinator.
+ *
+ * Proper Zigbee mode must be selected in Tools->Zigbee mode
+ * and also the correct partition scheme must be selected in Tools->Partition Scheme.
+ *
+ * Please check the README.md for instructions and more detailed description.
+ *
+ * Created by MikaFromTheRoof (https://github.com/MikaFromTheRoof)
+ */
+
+#ifndef ZIGBEE_MODE_ED
+#error "Zigbee end device mode is not selected in Tools->Zigbee mode"
+#endif
+
+#include "Zigbee.h"
+
+#define ZIGBEE_ILLUMINANCE_SENSOR_ENDPOINT 9
+uint8_t button = BOOT_PIN;
+uint8_t illuminance_sensor_pin = 6;  // Insert the analog pin to which the sensor (e.g. photoresistor) is connected
+
+ZigbeeIlluminanceSensor zbIlluminanceSensor = ZigbeeIlluminanceSensor(ZIGBEE_ILLUMINANCE_SENSOR_ENDPOINT);
+
+/********************* Illuminance sensor **************************/
+static void illuminance_sensor_value_update(void *arg) {
+  for (;;) {
+    // read the raw analog value from the sensor
+    int lsens_analog_raw = analogRead(illuminance_sensor_pin);
+    Serial.printf("[Illuminance Sensor] raw analog value: %d\r\n", lsens_analog_raw);
+
+    // conversion into zigbee raw illuminance value (typically between 0 in darkness and 50000 in direct sunlight)
+    // depends on the value range of the raw analog sensor values and will need calibration for correct lux values
+    // for demonstration purpose map the 12-bit ADC value (0-4095) to Zigbee illuminance range (0-50000)
+    int lsens_illuminance_raw = map(lsens_analog_raw, 0, 4095, 0, 50000);
```

Comment:
would it be nicer to set the 50,000 limit as a `#define ...` - it is used latter in the code.